src/output/gif.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace GOL\Output;
+
+use GifCreator\AnimGif;
+use GOL\Boards\Board;
+use Ulrichsg\Getopt;
+
+require_once "animgif/AnimGif.php";
+
+/**
+ * Saves the board as a gif.
+ *
+ * Use write() to write the Board data to the buffer.
+ * and flush() to write it on the disk.
+ */
+class Gif extends Output
+{
+    private $buffer = [];
+    private $delay = 1;
+    private $cellColor = [];
+    private $backgroundColor = [];
+    private $cellSize = 1;
+
+    /**
+     * Writes the current board to the Output.
+     *
+     * @param Board $_board Board to output.
+     */
+    public function write(Board $_board): void
+    {
+        $this->buffer[] = $_board->getGrid();
```

Comment:
Same here, you could have also written the gif to a file during write().
You would need space for temp-files but creating huge animations (imagine 1000+ Frames) would need less memory.